lib/logic/player/media_store_content_observer.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:android_content_provider/android_content_provider.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:sweyer/sweyer.dart';
+
+/// An event in [MediaStoreContentObserver.onChangeStream] stream.
+class MediaStoreContentChangeEvent {
+  final int id;
+  final int flags;
+  final Type contentType;
+
+  const MediaStoreContentChangeEvent({
+    required this.id,
+    required this.flags,
+    required this.contentType,
+  });
+}
+
+/// An Android MediaStore content observer.
+class MediaStoreContentObserver<T extends Content> extends ContentObserver {
+  MediaStoreContentObserver([this.contentType])
+    : assert(
+        Content.enumerate().contains(T),
+        "Specified type must be a subtype of Content",
+      );
+
+  final Type? contentType;
+
+  Stream<MediaStoreContentChangeEvent> get onChangeStream =>_changeSubject;
+  final _changeSubject = PublishSubject<MediaStoreContentChangeEvent>();
+
+  void register() {
+    AndroidContentResolver.instance.registerContentObserver(
+      uri: _uri,
+      observer: this,
+      // Can be uncommented for debug purposes - will notify about the changes the appitself
+      // made to the MediaStore.
+      //
+      // notifyForDescendants: true,
+    );
+  }
+
+  String get _uri => contentPick<T, String>(
+    contentType: contentType,
+    /// MediaStore.Audio.Media.EXTERNAL_CONTENT_URI
+    song: 'content://media/external/audio/media',
+    /// MediaStore.Audio.Albums.EXTERNAL_CONTENT_URI
+    album: 'content://media/external/audio/albums',
+    /// MediaStore.Audio.Playlists.EXTERNAL_CONTENT_URI
+    playlist: 'content://media/external/audio/playlists',
+    /// MediaStore.Audio.Artists.EXTERNAL_CONTENT_URI
+    artist: 'content://media/external/audio/artists',
+  );
+
+  @override
+  void dispose() {
+    AndroidContentResolver.instance.unregisterContentObserver(this);
+    _changeSubject.close();
+    super.dispose();
+  }
+
+  @override
+  void onChange(bool selfChange, String? uri, int? flags) {
```

Comment:
This is a copy of the native signature.

Reply:
Actually, it's not 1 to 1 copy, but it allows to distinguish between overloaded methods https://developer.android.com/reference/kotlin/android/database/ContentObserver

Reply:
What's the use of distinguishing them, they all do the same. The only reason why there are 3 different ones is because Java doesn't have default function parameters. There is no need to bring these idiosyncrasies from Java into Dart, since you have full control over this interface and nothing is forcing you to keep it exactly the same.

Reply:
I want to keep ability to track which of 3 overloads was called. Since this is an interface, I cannot just overthrow it and say that now Dart rules work here, because this would make the interface incomplete.

Reply:
Nah, actually thinking more of this, it really isn't useful at all to have null there - will change that